### DISCUSSION TITLE
Why is my subsampling extremely biased for high quality cells?

### DISCUSSION THREAD

**Discussion by markphillippebworth** [GreenleafLab/ArchR#785] [Questions / Documentation] [answered]
Hello!  I don't know if this is better submitted as a bug or discussion question. 

I've been running some reproducibility analysis by subsampling one particular cell type from one sample. I wrote my code so that the sampling should be random from the population, with ever increasing number of cells selected. However, when I look at the number of nFrags per cell, it looks the fewer cells I choose, the more biased they were towards have more nFrags per cells. My overall dataset has an average of about 4000 fragments per cell.

Here's what it looks like across iterations.
![image](https://user-images.githubusercontent.com/73675546/120049928-6e537c80-bfd0-11eb-92d9-e2fd77676e27.png)

Below is my code for subsetting.
```
cellRange <- c(seq(10,110, by = 20),seq(150,500, by = 50),seq(600,2400, by = 100))
#[1]   10   30   50   70   90  110  150  200  250  300  350  400  450  500  600
#[16]  700  800  900 1000 1100 1200 1300 1400 1500 1600 1700 1800 1900 2000 2100
#[31] 2200 2300 2400

for(i in 1:length(cellRange)){
CD4sets <- subsetArchRProject(CD4Test, getCellNames(CD4Test)[sample(cellRange[i]*2+22)], outputDirectory = "CD4sets/", force = TRUE)
saveArchRProject(CD4sets, "CD4sets/")
}
```

**Comment by rcorces** [comment, marked as answer]
Maybe its your (mis)usage of `sample`?
I believe you are supposed to provide a vector of values to sample from. so it should be
`sample(x = c(1:length(cellNames(CD4Test)), size = cellRange[i]*2+22)`
Otherwise, with the way you are using it, you randomly sample (without replacement) the numbers from 1 to `cellRange[i]*2+22`

for ex. 
```
> sample(5)
[1] 4 2 1 5 3
```

**Reply by markphillippebworth** [reply]
yep, that would do it! Thank you.
